security/tls-authorization/README.adoc

```diff
@@ -0,0 +1,226 @@
+. Download and start at least one Infinispan Server instance.
+
+== Unsigned Server certificates
+
+. Copy `server-keystore.pfx` to `$ISPN_HOME/server/conf`.
+. Add the `server-identities` configuration to the Infinispan Server security realm.
++
+Note that the default properties realm is removed.
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+    </ssl>
+  </server-identities>
+</security-realm>
+----
++
+. Create a cache named `secured` with any configuration.
++
+[source,xml]
+----
+<distributed-cache name="secured" mode="SYNC">
+  <encoding media-type="application/x-protostream"/>
+</distributed-cache>
+----
++
+. Run the client.
++
+[source]
+----
+$ export TEST=SIMPLE && mvn exec:exec
+----
+
+=== Unsigned Server certificates with client authorization
+
+. Copy `client-keystore.pfx` to `$ISPN_HOME/server/conf`.
+. Configure Infinispan Server to authenticate client certificates by adding `require-ssl-client-auth="true"` to the `endpoints` configuration.
++
+[source,xml]
+----
+<endpoints socket-binding="default"
+           security-realm="default"
+           require-ssl-client-auth="true"/>
+----
++
+. Add the `server-identities` configuration to the Infinispan Server security realm.
+. Include the `<truststore-realm/>` to authenticate certificates, which requires all public client certificates in the trust store.
++
+Note that the default properties realm is removed.
++
+[source,xml]
+----
+<security-realm name="default">
+  <server-identities>
+    <ssl>
+      <keystore path="server-keystore.pfx"
+                keystore-password="secret"
+                alias="infinispan-server"/>
+      <truststore path="client-truststore.pfx"
+                  password="ClientSecret"/>
+    </ssl>
+  </server-identities>
+  <truststore-realm/>
+</security-realm>
+----
++
+. Update the security authorization configuration for Infinispan Server to use the Common Name role mapper and define two client roles and associated permissions.
++
+In this example `Client1` and `Client2` correspond to the `cn` field in client certificates.
++
+[source,xml]
+----
+<cache-container>
+  <security>
+    <authorization>
+      <common-name-role-mapper/>
+      <role name="Client1" permissions="ALL"/>
+      <role name="Client2" permissions="READ"/>
+    </authorization>
+  </security>
+</cache-container>
+----
++
+. Create a cache with security authorization.
++
+[source,xml]
+----
+<distributed-cache name="secured" mode="SYNC">
+  <encoding media-type="application/x-protostream"/>
+  <security>
+    <authorization/>
+  <security>
+</distributed-cache>
+----
++
+. Run the clients.
++
+[source]
+----
+$ export TEST=SIMPLEAUTH && mvn exec:exec
+----
+
```

Comment:
I do think here it would be nice to say "Run git clean -fx" before moving on to the signed tutorial. That way you don't get keystores mixed up, etc.